src/transformers/models/big_bird/modeling_flax_big_bird.py

```diff
@@ -1597,14 +1616,14 @@ def enable_gradient_checkpointing(self):
             gradient_checkpointing=True,
         )
 
-    # Copied from transformers.models.bert.modeling_flax_bert.FlaxBertPreTrainedModel.init_weights
     def init_weights(self, rng: jax.random.PRNGKey, input_shape: Tuple, params: FrozenDict = None) -> FrozenDict:
         # init input tensors
         input_ids = jnp.zeros(input_shape, dtype="i4")
         token_type_ids = jnp.zeros_like(input_ids)
         position_ids = jnp.broadcast_to(jnp.arange(jnp.atleast_2d(input_ids).shape[-1]), input_shape)
         attention_mask = jnp.ones_like(input_ids)
         head_mask = jnp.ones((self.config.num_hidden_layers, self.config.num_attention_heads))
+        indices_prng_key = jax.random.PRNGKey(0)
 
         params_rng, dropout_rng = jax.random.split(rng)
         rngs = {"params": params_rng, "dropout": dropout_rng}
```

Comment:
Note that here we set the PRNG key for the indices to 0 each time. This means that the PRNG key for the indices is fixed.

We should **never** re-use PRNG keys in JAX. Doing so gives identical outputs, thus destroying the stochastic nature of the random permutation function. See https://jax.readthedocs.io/en/latest/jax-101/05-random-numbers.html#random-numbers-in-jax for more details.

Instead, we should generate new PRNG keys by _splitting_ the keys ourselves:

```suggestion
        rng, params_rng, dropout_rng, indices_rng = jax.random.split(rng, num=3)
        rngs = {"params": params_rng, "dropout": dropout_rng, "incides": indices_rng}
```